### PR TITLE
Fix lock icon by adding lock domain and default domain bug

### LINKED ISSUE #41: lock status shows as "?"
Hi,
I seem to be having a problem with the lock status for all of my locks.  If I add a mushroom lock card, the status icon is correct, it shows as Locked or Unlocked.  But in the strategy, it is a "?".

lock status issue

Correct Display:
lock correct

I also noticed that I'm unable to hide a domain.  The "title" option works, but it will not hide -- with our without a "title".

    views:
      cameras: false
    domains:
      default:
        hidden: true
        title: My Title
    areas:
      attic:
        hidden: true

### PR DESCRIPTION
**Pull request by AalianKhan**
Fixes #41

### CODE REVIEW

File: src/mushroom-strategy.js [hunk 1]
```diff
@@ -188,6 +188,7 @@ class MushroomStrategy {
       }
     }
 
+    // TODO: Check if default is hidden
```
Comment (AalianKhan) [inline src/mushroom-strategy.js:191]: Hey @DigiLive,
The miscellaneous domain doesn't check if the default domain is hidden. I was thinking we can just add an if statement to check if it's hidden but we can also move it up line 177 and use the domain variable to check if it's hidden. What's your call?

File: src/cards/MediaPlayerCard.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ class MediaPlayerCard extends AbstractCard {
    */
   #defaultOptions = {
     type: "custom:mushroom-media-player-card",
+    icon: undefined,
```
Comment (DigiLive) [inline src/cards/MediaPlayerCard.js:20]: Why added, the docs says it's optional or else why `undefined`?

File: src/cards/LockCard.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * Used to create a card for controlling an entity of the lock domain.
+ *
+ * @class
+ * @extends AbstractCard
+ */
+class LockCard extends AbstractCard {
+  /**
+   * Default options of the card.
+   *
+   * @type {lockCardOptions}
```
Comment (DigiLive) [inline src/cards/LockCard.js:15]: `lockCardOptions` should be added to file `src\cards\typedefs.js`

File: src/mushroom-strategy.js (hunk 1 above)
Comment (DigiLive) [reply src/mushroom-strategy.js:191]: This is a separate bug and you should create another issue or PR for it.
Try to keep each issue or "topic of change" separate from each other.

E.g.: 1 to add the Lock-Card, 1 to fix the MediaPlayer-Card, 1 to fix Miscellaneous.

File: src/cards/MediaPlayerCard.js (hunk 2 above)
Comment (AalianKhan) [reply src/cards/MediaPlayerCard.js:20]: If it is not added, the icon used is set by the abstract card which is `mdi:help-circle`. 
When it is set to undefined, the icon used is the entity id's icon.

File: src/cards/LockCard.js (hunk 3 above)
Comment (AalianKhan) [reply src/cards/LockCard.js:15]: not sure what to add because there aren't any options

Comment (DigiLive) [reply src/cards/LockCard.js:15]: When having no options in addition to the options of the Abstract Card...

```js
/**
 * @typedef {abstractOptions} LockCardOptions Lock Card options.
 * @memberOf typedefs.cards
 */
```

**Review by DigiLive — APPROVED** [review]

**Pull request merged**